openjdk-integ-tests/src/test/java/libcore/javax/net/ssl/SSLSocketTest.java

```diff
@@ -1649,7 +1658,8 @@ private void test_SSLSocket_interrupt_case(boolean readUnderlying, boolean close
             final InputStream inputStream = toRead.getInputStream();
             @SuppressWarnings("unused")
             int value = inputStream.read();
-            fail();
+            // FD-based socket on OpenJDK seems to return -1 here.
+            assertEquals(-1, value);
```

Comment:
I guess this is where the FD-based socket has a problem. It's not getting the notification from OpenJDK's infrastructure from this thread. The correct behavior is to throw `SocketException`, but this isn't working because it needs that infrastructure support.

Maybe we should assert some condition on either side of the catch. Like `assertUsingEngineBasedSocket(true);`